tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContent.scala

```diff
@@ -0,0 +1,82 @@
+package com.linagora.tmail.encrypted
+
+import com.google.common.io.ByteSource
+import com.linagora.tmail.pgp.Encrypter
+import org.apache.james.jmap.api.model.Preview
+import org.apache.james.mailbox.model.{MessageId, ParsedAttachment}
+
+import java.io.ByteArrayOutputStream
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicInteger
+import scala.jdk.OptionConverters._
+
+class EncryptedEmailContentFactory(encrypter: Encrypter) {
+
+  def encrypt(clearEmailContent: ClearEmailContent, messageId: MessageId): EncryptedEmailContent =
+    EncryptedEmailContent(encryptedPreview = encryptPreview(clearEmailContent.preview),
+      encryptedHtml = encrypt(clearEmailContent.html),
+      hasAttachment = clearEmailContent.hasAttachment,
+      encryptedAttachmentMetadata = encryptAttachmentMetadata(clearEmailContent.attachments, messageId),
+      encryptedAttachmentContents = encryptAttachmentContent(clearEmailContent.attachments))
+
+  def encryptPreview(preview: Preview): String =
+    encrypt(preview.getValue)
+
+  def encryptAttachmentMetadata(parsedAttachments: List[ParsedAttachment], messageId: MessageId): Option[String] = {
+    if (parsedAttachments.isEmpty) return None
```

Comment:
You should use the pattern case matching syntax here :)

Reply:
Sometimes, I feel confused between the two options. When I tried to use a matching pattern or monad, somebody told me I should use if-else to understand easily. And When I tried to if-else, somebody said to me should use functional style. :(

Reply:
Ahah yeah but on this case I think a matching pattern is better sorry^^'